src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -116,9 +121,10 @@ private BoundExpression BindDeconstruction(AssignmentExpressionSyntax node, Diag
                 var type = boundRHS.Type ?? voidType;
                 return new BoundDeconstructionAssignmentOperator(
                             node,
-                            DeconstructionVariablesAsTuple(node, checkedVariables, diagnostics, hasErrors: true),
+                            DeconstructionVariablesAsTuple(node, checkedVariables, resultIsUsed ? diagnostics : null, hasErrors: true),
```

Comment:
Is there a test for this case, where `boundRHS.Type.IsErrorType()` and `resultIsUsed == false`?